[jsk_tools] List jsk Github repos

wkentaro opened the pull request:
Depends on https://github.com/ros/rosdistro/pull/13011

wkentaro commented:
Usage

```bash
rosrun jsk_tools list_repos.py > jsk_all.rosinstall
```

wkentaro commented:
Please merge this.
<img src="https://cdn.meme.am/instances/400x/65467812.jpg" width="300px" />

k-okada commented:
if you'll like to crate something like below, that's very nice, And next
question is how to review these PR, if you do not and ask someone for
review, please read hrp2016 update
https://github.com/jsk-ros-pkg/jsk_demos/pull/1203,
https://github.com/jsk-ros-pkg/jsk_pr2eus/pull/237#issuecomment-242899413
 and https://github.com/euslisp/EusLisp/issues/203#issuecomment-256221070
discussion
and learn how people reacts.

did you consider another option? For example update https://github.com/jsk-
ros-pkg/jsk-ros-pkg.github.io to generate ros install file?

Please do not paste list of private repos, it might contain an information
which should not be disclosed

Please be very careful when you use pip

Why do we have to merge this NOW? Is someone waiting for this?

I sent PR for fixing that all jsk-ros-pkg and start-jsk packages.

   - jsk-ros-pkg/jsk-ros-pkg-unreleased#73
   <https://github.com/jsk-ros-pkg/jsk-ros-pkg-unreleased/pull/73>
   - jsk-ros-pkg/euslib#219 <https://github.com/jsk-ros-pkg/euslib/pull/219>
   - jsk-ros-pkg/jsk_visualization#644
   <https://github.com/jsk-ros-pkg/jsk_visualization/pull/644>
   - jsk-ros-pkg/jsk_smart_apps#68
   <https://github.com/jsk-ros-pkg/jsk_smart_apps/pull/68>
   - #1917 <https://github.com/jsk-ros-pkg/jsk_recognition/pull/1917>
   - jsk-ros-pkg/jsk_demos#1204
   <https://github.com/jsk-ros-pkg/jsk_demos/pull/1204>
   - jsk-ros-pkg/jsk_control#637
   <https://github.com/jsk-ros-pkg/jsk_control/pull/637>


2016年10月26日水曜日、Kei Okada<k-okada@jsk.t.u-tokyo.ac.jp
<javascript:_e(%7B%7D,'cvml','k-okada@jsk.t.u-tokyo.ac.jp');>>さんは書きました:

> please do not threaten me
>
>
> 2016年10月26日水曜日、Kentaro Wada<notifications@github.com>さんは書きました:
>
>> Please merge this.
>>
>> <https://camo.githubusercontent.com/aa3c5ac3f86af08438376fe11bad62cd85779f61/68747470733a2f2f63646e2e6d656d652e616d2f696e7374616e6365732f343030782f36353436373831322e6a7067>
>>
>> —
>> You are receiving this because you are subscribed to this thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/jsk-ros-pkg/jsk_common/pull/1463#issuecomment-256243988>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe-auth/AAeG3ChanaDdqNYqQwdR2_pQ0wGZFYH7ks5q3tC8gaJpZM4Kdf0v>
>> .
>>
>
>
> --
> --
> ◉ Kei Okada
>
>
>

-- 
--
◉ Kei Okada

k-okada commented:
please do not threaten me


2016年10月26日水曜日、Kentaro Wada<notifications@github.com>さんは書きました:

> Please merge this.
>
> <https://camo.githubusercontent.com/aa3c5ac3f86af08438376fe11bad62cd85779f61/68747470733a2f2f63646e2e6d656d652e616d2f696e7374616e6365732f343030782f36353436373831322e6a7067>
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/jsk-ros-pkg/jsk_common/pull/1463#issuecomment-256243988>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAeG3ChanaDdqNYqQwdR2_pQ0wGZFYH7ks5q3tC8gaJpZM4Kdf0v>
> .
>


-- 
--
◉ Kei Okada

wkentaro commented:
> please do not threaten me

Sorry about that. I will refrain from that.

> if you'll like to crate something like below, that's very nice, And next
question is how to review these PR, if you do not and ask someone for
review, please read hrp2016 update
jsk-ros-pkg/jsk_demos#1203,
jsk-ros-pkg/jsk_pr2eus#237 (comment)
 and euslisp/EusLisp#203 (comment)
discussion
and learn how people reacts.

>> 結局、それほど必要なかったのでは？と思ったりします．

と言われないように自分が使い続けやすい環境（masterブランチにマージされた状態）
を作ろうとしてしまうというのがあります。（結局branchが散在すると自分で作っても使わなくなるので）
つまり、"Please merge this" ではなくて、 "Please review this"にすべきという話でしょうか？

> did you consider another option? For example update https://github.com/jsk-
ros-pkg/jsk-ros-pkg.github.io to generate ros install file?

Actually, no. I'd like to add the `list_repo.py` lists ros packages using github api like `jsk_tools/git_commit_alias.py` for users.

> Please do not paste list of private repos, it might contain an information
which should not be disclosed

Sorry about that. I removed them.

> Please be very careful when you use pip

So I replaced current `pygithub3` with `PyGithub`.

> Why do we have to merge this NOW? Is someone waiting for this?

No, there's no necessity to merge this **NOW**.

k-okada commented:
> No, there's no necessity to merge this NOW.

that's good to know, the skull/skeleton tole me like "PLEASE DO IT __NOW__"
